ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalPropertyParser.java

```diff
@@ -55,9 +53,21 @@ public Object parse(final String property, final Object value) {
         .orElseThrow(() -> new IllegalArgumentException(String.format(
             "Not recognizable as ksql, streams, consumer, or producer property: '%s'", property)));
 
-    final Object parsedValue = configItem.parseValue(value);
+    final Object parsedValue = parse(value, configItem);
 
     validator.validate(configItem.getPropertyName(), parsedValue);
     return parsedValue;
   }
+
+  private Object parse(final Object value, final ConfigItem configItem) {
+    final boolean isShortProperty = configItem.getPropertyType()
+        .map(type -> type == Type.SHORT)
+        .orElse(false);
+
+    if (isShortProperty) {
```

Comment:
why not just always do `configItem.parseValue(String.valueOf(value))`?

Reply:
+1, if we know that strings can be safely coerced to the underlying type, why not only pass strings? At the very least, we should document the reason why short is special cased and not other types. Is short the only one which can't be coerced?

Reply:
Two reasons:
- Why pay the price of Object -> String -> Object.
- Why introduce potentially new issues with the way ConfigDef parses Strings?

@apurvam - the documentation you request is already there. check the comment on the line below.

Reply:
@big-andy-coates the price? it isn't exactly on the critical path - so i don't see it as an issue at all. We already know that ConfigDef parses Strings - that is what it generally does...

Reply:
True.  OK, convinced.